napari/components/dims.py

```diff
@@ -201,14 +166,12 @@ def ndim(self, ndim):
         if cur_ndim == ndim:
             return
         elif ndim > cur_ndim:
-            # Range value is (min, max, step) for the entire slider
-            self._range = [(0, 2, 1)] * (ndim - cur_ndim) + self._range
-            # Point is the slider value if in point mode
-            self._point = [0] * (ndim - cur_ndim) + self._point
-            # Interval value is the (min, max) of the slider selection
-            # if in interval mode
-            self._interval = [(0, 1)] * (ndim - cur_ndim) + self._interval
-            self._mode = [DimsMode.POINT] * (ndim - cur_ndim) + self._mode
+            for i in range(ndim - cur_ndim):
+                # Range value is (min, max, step) for the entire slider
+                self.range.insert(0, (0, 2, 1))
+                # Point is the slider value
+                self.point.insert(0, 0)
+
             self._order = list(range(ndim - cur_ndim)) + [
```

Comment:
Have to think about how to do things like this with the evented list - i probably want to modify in place rather than create/destroy like with the `+` syntax, but thanks for adding to #1444

Reply:
do what exactly? prepend a list to a list?  There needn't be anything different about the evented list from a "regular" list (though we may need to update a couple magic methods).  What syntax do you want to use?  I think you have three options:
1. use list slice insertions syntax: `ev_list[:0] = [x,x,x,...]`  (currently, this will emit one changed even but we could make it emit as many as there are insertions)
2. use for loop with `ev_list.insert(0, ...)` (this will emit one `changed` event for each insertion)
3. we _could_ modify the `__radd__` method on the evented list to prepend when you do `x = [] + ev_list`, but that would be breaking standard python practice (i.e. that always returns a new list object)

Reply:
ooo `use list slice insertions syntax` this is quite nice, otherwise I was just going to do a for loop. I probably would have expected the slice insertion to emit as many events as insertions, but I guess I expect the `clear` to probably just emit one, so maybe it's ok.

main point was to note not to create/ destroy as I had been doing. I would not modify `__radd__`

Reply:
if you find that you _need_ it to emit that many events, it would be easy... we'd just need to check whether `value` was iterable and iterate through it instead of passing on the slice syntax to the private list.
https://github.com/napari/napari/blob/7bac12ddb0e26d54006d686c599152162de242e5/napari/utils/events/containers/_list.py#L131-L135

> but I guess I expect the clear to probably just emit one, so maybe it's ok.

for synchronization with `QAbstracItemModels`, it ends up being easier to have there be a 1:1 correspondence between changing items and events... but it's not strictly necessary.  so if you run into any issue with clear emitting an event for each item being removed from the list, let me know.